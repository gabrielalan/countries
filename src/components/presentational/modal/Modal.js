import React from 'react';
import { createPortal } from 'react-dom';
import PropTypes from 'prop-types';

import Icon from '../icon/Icon';

export default function Modal({ title, children, onClose, modalRoot }) {
  const container = <div className="modal">
    <div className="modal__content">
      <div className="modal__content__title">
        <h1>{title}</h1>
        <button type="button" className="btn btn--small btn--primary--outline btn--iconed" onClick={onClose}>
          <Icon id="clear" className="btn__icon" />
        </button>
      </div>
      <div className="modal__content__body">
        {children}
      </div>
    </div>
  </div>;

  return createPortal(container, modalRoot);
}

Modal.defaultProps = {
  modalRoot: window.document.body
};

Modal.propTypes = {
  title: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
};
