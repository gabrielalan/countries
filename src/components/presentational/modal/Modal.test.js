import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Modal from './Modal';

jest.mock('react-dom');

describe('Modal', () => {
  beforeAll(() => {
    ReactDOM.createPortal.mockImplementation(element => element);
  });

  it('should render modal correctly', () => {
    const spy = jest.fn();
    const tree = renderer.create(<Modal title="Title" onClose={spy}>
      <p>children</p>
    </Modal>);

    renderer.act(() => {
      tree.root.findByType('button').props.onClick();
    });
    
    expect(spy).toHaveBeenCalled();
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
