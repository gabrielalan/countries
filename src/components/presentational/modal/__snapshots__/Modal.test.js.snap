// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render modal correctly 1`] = `
<div
  className="modal"
>
  <div
    className="modal__content"
  >
    <div
      className="modal__content__title"
    >
      <h1>
        Title
      </h1>
      <button
        className="btn btn--small btn--primary--outline btn--iconed"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        type="button"
      >
        <span
          className="icon btn__icon"
        >
          <svg
            aria-hidden="true"
            className="icon__svg"
            viewBox="0 0 10 10"
          >
            <use
              xlinkHref="#icon-clear"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      className="modal__content__body"
    >
      <p>
        children
      </p>
    </div>
  </div>
</div>
`;
